broadinstitute/single_cell_portal_core
# Show reference expression by tissue in related genes ideogram (SCP-5486)

This helps users investigate the typical expression of any gene, using data from [GTEx](https://gtexportal.org), a large reference dataset.  

### Overview
Here are some general questions users can easily answer via this enhancement in related genes ideogram:
* In which tissues is my gene typically most expressed?
* Is expression usually concentrated in one tissue, or across many?
* How does the expression of this gene in this sample compare to that in a large, representative cohort?
* Within each tissue, is expression usually normally distributed, skewed, or multimodal?
* What is the typical median, maximum, and minimum expression?

### Video
Here's how it looks!

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/36daa79a-9d8e-4e50-ac56-acca9aaa9bc4

### Test
To manually test:
1.  `yarn install`
2.  Go to any study with human genes
3.  Search for a gene
4.  Hover over a gene in related genes ideogram
5.  In the ideogram tooltip, confirm you see expression distribution curves for 3 tissues
6.  Click "More..." in the tooltip
7.  Confirm 10 curves are shown
8.  Hover over various curves
9.  Upon hovering over each curve:
  a.  Confirm the mini-curves rescale to use the focused tissue as a reference perspective
  b.  Confirm a detailed curve for the focused tissue is displayed where the molecular diagrams usually are

### Further details
This satisfies [SCP-5486](https://broadworkbench.atlassian.net/browse/SCP-5486).  Integrating this upstream enhancement from related genes ideogram was done as part of Developer's Choice Days.  More context is available in the [SCP demo video](https://drive.google.com/file/d/1e6rkAFRVbIMGj7cHb90izikCtulwIFy_/view?t=13m9s) from 2024-01-16.

[SCP-5486]: https://broadworkbench.atlassian.net/browse/SCP-5486?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
